src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalInitiallyNeeded.tsx

```diff
@@ -13,36 +14,60 @@ import { useAppDispatch, useAppSelector } from 'state/store'
 import { useIsApprovalNeeded } from './useIsApprovalNeeded'
 
 const useIsApprovalInitiallyNeededForHop = (
+  tradeQuoteId: string | undefined,
   tradeQuoteStep: TradeQuoteStep | undefined,
   sellAssetAccountId: AccountId | undefined,
 ) => {
   const [isApprovalInitiallyNeeded, setIsApprovalInitiallyNeeded] = useState<boolean | undefined>()
+  const [isAllowanceResetNeeded, setIsAllowanceResetNeeded] = useState<boolean | undefined>()
 
-  const { isLoading, isApprovalNeeded } = useIsApprovalNeeded(tradeQuoteStep, sellAssetAccountId)
+  const {
+    isLoading,
+    isFetching,
```

Comment:
blocking: using `isFetching` causes subsequent calls to `setTradeInitialized()` in `MultiHopTradeConfirm.tsx` which will hide the hop content after being on the allowance step and a refetch happens.

I am validating this logic, but I don't understand why all of the extra state is required here. Is there multi hop concerns?

Testing the following which is looking good so far, but will have to turn on lifi to validate multi hop if the edge case lies there:

```ts
  const { isLoading, data: isApprovalNeededData } = useIsApprovalNeeded(
    tradeQuoteStep,
    sellAssetAccountId,
  )

  useEffect(() => {
    if (isApprovalNeededData?.isApprovalNeeded === undefined) return
    setIsApprovalInitiallyNeeded(isApprovalNeededData.isApprovalNeeded)
  }, [isApprovalInitiallyNeeded, isApprovalNeededData, isLoading])

  useEffect(() => {
    if (isApprovalNeededData?.isAllowanceResetNeeded === undefined) return
    setIsAllowanceResetNeeded(isApprovalNeededData?.isAllowanceResetNeeded)
  }, [isAllowanceResetNeeded, isApprovalNeededData, isLoading])

  const result = useMemo(
    () => ({
      isLoading,
      isApprovalInitiallyNeeded,
      isAllowanceResetNeeded,
    }),
    [isLoading, isApprovalInitiallyNeeded, isAllowanceResetNeeded],
  )

  return result
```

Reply:
Ok, while it looked "ok" at first take, I see the fact that initial state is not saved in my suggestion and the changing of trade routes requires the reset.

So just the `isFetching` is blocking, but another question would be race conditions between the `useEffects`. It seems like there could be a chance that the reset is run after the  set approval/allowance which would show the wrong state until the next refetch of allowance occurs. Is this accurate and would there be a way to handle the reset in the same effect based on `tradeQuoteId` then to ensure that isn't possible?

Reply:
Nicely spotted, `isFetching`  issue fixed in 0a91518d99dee04723c422b33539f9dd099c3d95

Reply:
Have moved the reset useEffect above the setters to eliminate theoretical possibility of weirdness when quotes change 954da8f7766a3a7f39410c196d31b4ea103bb4a5